src/modules/UI/Wallets/action.js

```diff
@@ -93,40 +111,247 @@ export const setEnabledTokens = (walletId: string, enabledTokens: Array<string>,
   const wallet = CORE_SELECTORS.getWallet(state, walletId)
   // now actually tell the wallet to enable the token(s) in the core and save to file
   WALLET_API.setEnabledTokens(wallet, enabledTokens, disabledTokens)
-  .then(() => {
+  .then((enabledTokens) => {
     // let Redux know it was completed successfully
     dispatch(setTokensSuccess())
+    dispatch(updateWalletEnabledTokens(walletId, enabledTokens))
     // refresh the wallet in Redux
     dispatch(refreshWallet(walletId))
   })
-  .catch((e) => console.log(e))
+  .catch((e) => dispatch(displayErrorAlert(e.message)))
 }
 
 export const getEnabledTokens = (walletId: string) => (dispatch: Dispatch, getState: GetState) => {
```

Comment:
You are calling getEnabledTokens for all wallets including those that aren't ETH. This will throw an error for BTC, LTC, Dash wallets as they don't support tokens and you are calling addCustomToken() on them.

Worst yet, you are calling addCustomToken on all wallets for all customTokens that you have in the account, even if the wallet doesn't have that token enabled. This will be a big performance and resource hit as each ETH wallet will have to query for every token even if it's not enabled in that wallet.